### PR TITLE
feat: add coordinator merge-watch routing

### LINKED ISSUE #359: Auto-merge slice 3/4: Coordinator merge-watch — pending-state classification, Fixer/re-Triage routing (oracle review required)
## Parent

#352

## What to build

Tracer-bullet slice 3 of 4 for the Reviewer auto-merge PRD. After slice #358, Reviewer opts approved Looper PRs into GitHub-native auto-merge but **nothing watches what happens next**. GitHub may merge, or it may stall on red CI, conflicts, branch-protection changes, transient API errors, or human override. This slice adds Coordinator's per-tick merge-watch — preserving ADR-0001's stateless property by rooting the watch on tracked Issues (Coordinator's existing scope) rather than on long-lived Reviewer loops.

This slice is **authority-bearing**. **Mandatory `@oracle` review per AGENTS.md applies — Coordinator gains its first responsibility for reading PR state and routing failures back to Issue-level actions (re-Triage). The boundary preservation per ADR-0003 is critical: Coordinator's target of action remains the Issue (label removal triggering re-Triage) or a Trigger label spawning Fixer; Coordinator does not directly modify the PR.**

The implementation PR description must explicitly request `@oracle` review and cite ADR-0001, ADR-0003, ADR-0005.

**Coordinator PR-mutation boundary.** Coordinator normally reads PRs and acts on the linked Issue, but this slice has one explicit PR-mutation exception: Coordinator may add only the configured Fixer Trigger label to the watched PR as a routing signal for `Conflict` / `RedCI`. Coordinator must not otherwise mutate PRs: no PR comments, no merge/close/update-branch operations, no review state changes, and no non-Fixer labels.

Concretely:

- **Deep module: `internal/coordinator/mergewatch`.** Pure function from `(PRSnapshot, PriorWatchMarker, RetryBudget) → WatchAction`. The `PriorWatchMarker` is the most-recent `<!-- looper:coordinator:merge-watch retries=N -->` marker comment on the linked Issue (or absent for first observation). The function classifies the PR's merge-pending state and returns one of:
  - `Merged` — GitHub has merged the PR. No-op (the existing Issue-close flow handles downstream).
  - `StillPending` — auto-merge enabled, mergeability is known, no failure observed, GitHub is waiting on its own gates. No-op.
  - `Indeterminate{firstUnknownAt, deadlineExceeded}` — GitHub has not computed mergeability yet (`mergeable=null` / `mergeable_state="unknown"`, commonly after a fresh push or base movement). Action before deadline: no label mutation, do **not** consume transient retry budget, and persist/keep `first_unknown_at` in the marker comment so the bound survives Coordinator restarts. Action after deadline: treat as `BranchProtectionChanged` semantics (return the linked Issue to re-Triage and remove the marker).
  - `Conflict` — merge conflict detected. Action: spawn Fixer on the PR via the existing Fixer Trigger label.
  - `RedCI` — required check failed. Action: spawn Fixer on the PR.
  - `BranchProtectionChanged` — a previously-approved PR no longer satisfies branch protection (new required CODEOWNERS, new required check the PR doesn't satisfy). Action: remove `triaged` and `dispatch/*` from linked Issue (returns to re-Triage candidate set).
  - `HumanDisabledAutoMerge` — a human clicked GitHub's "Disable auto-merge" button. Action: stop watching this PR; remove the `<!-- looper:coordinator:merge-watch -->` marker comment; no other change. Silent override-respect.
  - `TransientError{retriesLeft, suggestedDelay}` — temporary GitHub/API failure, including 5xx, network timeout, 429, and 403 secondary-rate-limit / abuse-detection responses. Action: no-op this tick; the marker comment's retry counter is decremented (or initialized from `transientRetries` config on first error). The gateway must thread GitHub retry hints (`Retry-After`, `x-ratelimit-reset`) into `suggestedDelay`; for 5xx/timeouts it may provide the runner's default backoff hint.
  - `TransientError{retriesLeft=0}` — retry budget exhausted. Action: treat as `BranchProtectionChanged` (return to re-Triage).
  
  The module is pure: zero GitHub I/O. PR snapshot, marker state, and retry budget are inputs.

- **Marker comment design (preserves ADR-0001).** The retry counter, watched PR identity, watched head SHA, mergeability-unknown bound, and rate-limit/backoff delay are persisted in the marker comment body itself: `<!-- looper:coordinator:merge-watch pr=N head_sha=<sha> retries=N first_unknown_at=<RFC3339-or-empty> next_retry_at=<RFC3339-or-empty> -->`, edited in place across ticks. Public, durable, idempotent. The marker lives on the **linked Issue** (Coordinator's scope), not on the PR — keeping Coordinator's target of action Issue-rooted per ADR-0003. The marker is removed on `Merged` and on `HumanDisabledAutoMerge`; persists across `StillPending` ticks unchanged; updated on first `Indeterminate` to set `first_unknown_at`; updated on `TransientError` to track the decremented counter and `next_retry_at`; removed when `BranchProtectionChanged` action returns the Issue to re-Triage (re-Triage starts fresh). If mergeability later becomes known, clear `first_unknown_at` on the next marker update so future unknown periods get a fresh deadline. If the watched PR number no longer matches the marker, treat the marker as stale and remove it before starting fresh. If the watched head SHA changes, reset retries, clear `first_unknown_at`, clear `next_retry_at`, and continue watching the new head as a fresh attempt.

- **Indeterminate bound.** `Indeterminate` has its own wall-clock bound separate from transient retry budget, configured as `maxIndeterminateDuration` (default: 15 minutes). The bound is anchored at marker `first_unknown_at`, not process memory, so Coordinator restarts do not reset it. Before the deadline, no label or retry-counter mutation occurs except creating/updating the marker to persist `first_unknown_at`; after the deadline, route using `BranchProtectionChanged` semantics because Coordinator cannot establish that the PR is still on a valid auto-merge path. `first_unknown_at` is scoped to `(pr, head_sha)` and resets on head SHA change.

- **Retry-delay durability.** `TransientError.suggestedDelay` is persisted as marker `next_retry_at`. If `now < next_retry_at`, the runner skips PR polling and marker mutation for that watch item and does not decrement retries. `next_retry_at` is scoped to `(pr, head_sha)` and resets on head SHA change.

- **Marker concurrency rule.** Merge-watch application must be serialized per linked Issue before editing/deleting the marker comment. The primary v1 mechanism is the existing Coordinator per-Issue execution guard if it covers merge-watch; otherwise add a runner-level keyed mutex by `(repo, issue_number)`. Do **not** rely on GitHub issue-comment `PATCH` as compare-and-swap: it can return 200 while clobbering a concurrent update. The slice must not allow two overlapping ticks to both read `retries=3` and both write `retries=2` silently.

- **Modify `internal/coordinator/runner` per-tick orchestration.** For every tracked Issue with a Trigger label and an open Looper PR carrying GitHub auto-merge enabled (the `auto_merge` field is non-null on the PR):
  1. If a marker has `next_retry_at` in the future, skip this watch item without polling or decrementing retries. Otherwise read the PR snapshot via the gateway (PR state, head SHA, mergeable state, required-check status, `auto_merge` field, and auto-merge actor/authority).
  2. Read the prior watch marker from the linked Issue's comments (one extra comment-list call per watched Issue per tick).
  3. Call `mergewatch.Classify` with snapshot + marker + retry budget.
  4. Apply the returned `WatchAction` in the locked order:
     - `Merged`: remove the marker comment; no other action (Issue close drives downstream via PRD #351 dependency gate)
     - `StillPending`: no action this tick
     - `Indeterminate`: before deadline, no label mutation and no retry decrement; create/update the marker only as needed to persist `first_unknown_at`; after deadline, apply `BranchProtectionChanged` semantics
     - `Conflict` / `RedCI`: apply the Fixer Trigger label (`looper:fix` or whatever is configured in `roles.fixer.triggers.labels`); update the marker comment to reflect the action taken with a brief one-line summary in the comment body (above the HTML-marker line)
     - `BranchProtectionChanged` (and exhausted-retries `TransientError`, and Indeterminate deadline exceeded): remove `triaged`, `dispatch/*`, and stale role trigger labels from the linked Issue; remove the marker comment
     - `HumanDisabledAutoMerge`: remove the marker comment; no other action
     - `TransientError` with retries left: edit the marker comment in place to update the retry counter and `next_retry_at`
  
  The Fixer-trigger-label-application is the existing label-driven Role pattern from PRD #334 — no new mechanism. Coordinator's authority over the Issue ends at applying the Trigger label; Fixer's existing reactive loop discovers and acts.

- **Identifying watched PRs from the Issue side.** The runner enumerates tracked Issues (existing Coordinator candidate set extended to "Issues with a Trigger label applied or a merge-watch marker present"). For each, the gateway resolves linked PRs via `GET /repos/{}/{}/issues/{n}/timeline` or the existing PR-Issue linkage (already used in `gateway.go:2203-2249` per `closedByPullRequestsReferences`). A PR is watched if either:
  1. the linked Issue already carries a merge-watch marker for that PR number, or
  2. there is exactly one eligible linked Looper PR carrying a `looper:` label and GitHub auto-merge is enabled by Looper.

  `auto_merge` non-null by itself is not authority. The snapshot must prove Looper ownership of the auto-merge opt-in, either via `auto_merge.enabled_by` matching the Looper bot identity or another explicit Looper authority signal established by slice #357. PRs without Looper-owned auto-merge enabled and without an existing marker are out of scope — they were either Refused at OptIn time (stamped refusal comment exists, awaiting human merge) or are human-authored. If an Issue has multiple eligible linked PRs and no marker disambiguates the watched PR, v1 performs no mutation, logs/warns an ambiguous-linkage event, and leaves the marker absent.

- **PR snapshot contract.** `mergewatch.Classify` receives an explicit `PRSnapshot` rather than raw GitHub responses. It must include at least: repo, PR number, linked Issue number, head SHA, merged/merged_at, PR open/closed state, `auto_merge` enabled/null, auto-merge actor/Looper authority signal, `mergeable`, `mergeable_state` or equivalent, required-check conclusion summary, failing required-check names, and gateway-classified temporary error metadata (`suggestedDelay` where present). Classification precedence is: `Merged` → `HumanDisabledAutoMerge` for a previously watched Looper-owned PR whose auto-merge is now null → `Conflict` → `RedCI` → `StillPending` for known expected progress → `Indeterminate` for mergeability unknown → `BranchProtectionChanged` for known unsatisfied branch-protection drift.

- **Gateway wrappers in `internal/infra/github/gateway`** (pure I/O):
  - `GET /repos/{}/{}/pulls/{}` exposing `auto_merge`, `mergeable`, `mergeable_state` fields
  - `GET /repos/{}/{}/commits/{ref}/check-runs` exposing required-check status (for RedCI classification)
  - Comment-edit wrapper for in-place updates of the marker comment (`PATCH /repos/{}/{}/issues/comments/{id}`); marker mutation safety is provided by per-Issue serialization in the Coordinator runner, not by GitHub comment PATCH semantics
  - Comment-delete wrapper for marker removal (`DELETE /repos/{}/{}/issues/comments/{id}`)

- **Unit tests for `mergewatch`** (decision-matrix driven, table-style; all seven WatchAction outcomes plus boundary cases):
  - PR `merged=true` → `Merged`
  - PR auto-merge enabled, mergeable, no failed checks → `StillPending`
  - PR `mergeable=null` / `mergeable_state="unknown"` before `maxIndeterminateDuration` → `Indeterminate` with no retry decrement and `first_unknown_at` persisted
  - PR `mergeable=null` / `mergeable_state="unknown"` after `maxIndeterminateDuration` → `BranchProtectionChanged` semantics
  - PR `mergeable_state="dirty"` (conflict) → `Conflict`
  - PR `mergeable_state="blocked"` with required check failed → `RedCI`
  - PR auto-merge enabled but new required check appears that PR doesn't satisfy → `BranchProtectionChanged`
  - PR `auto_merge=null` after previously being non-null (human disabled) → `HumanDisabledAutoMerge`
  - 5xx, timeout, 429, and 403 secondary-rate-limit / abuse-detection responses classified by upstream gateway → `TransientError` with correct decrement and `suggestedDelay` threaded from `Retry-After` / `x-ratelimit-reset` where present
  - `TransientError` with `retriesLeft=0` → routes to `BranchProtectionChanged` semantics
  - Marker absent + first observation of any failure state → `TransientError` initialized to full retry budget (so transient errors at first observation don't immediately escalate)

- **Unit tests for the `runner` merge-watch orchestration** (table-driven):
  - For each `WatchAction` outcome, verify the runner makes exactly the GitHub calls specified by the locked action ordering
  - `Indeterminate` performs no label mutation and no retry decrement; it only creates/updates the marker when needed to persist `first_unknown_at`, and routes to re-Triage after its deadline
  - Marker comment edit-in-place vs delete-vs-create idempotency: re-running on the same input twice produces no second-run side effects
  - Marker comment concurrency: two overlapping observations cannot silently lose a retry decrement; per-Issue serialization prevents overlapping marker mutation for the same linked Issue
  - Marker-first discovery: merged and human-disabled PRs are still cleaned up even when `auto_merge` is now null or the PR is no longer in the normal discovery predicate
  - Head SHA changes reset retries, `first_unknown_at`, and `next_retry_at`
  - Multiple eligible linked PRs without a marker produce no mutation and log/warn ambiguity

- **Two `fake-gh` integration tests**:
  1. **Conflict path**: PR is auto-merge-pending; tick 1 detects conflict; runner applies Fixer Trigger label; marker comment is updated with action summary. Verified: Fixer Trigger label appears on the PR.
  2. **Human-disabled path**: PR is auto-merge-pending with marker present; human disables auto-merge between ticks (`auto_merge` becomes null); tick 2 detects, removes the marker comment, makes no other change. Verified: marker comment is gone; no Fixer label applied; no re-Triage label changes.

This slice does NOT add the concurrency pre-emption check (slice 4, #TBD-B4) or CONTEXT.md updates (slice 4). The merge-watch is functional end-to-end after this slice; slice 4 closes out the rough edges.

Use domain language from `CONTEXT.md` consistently — including the Coordinator/Sweeper boundary discipline from ADR-0003.

## Acceptance criteria

- [ ] `internal/coordinator/mergewatch` exists as a pure-function deep module classifying PR auto-merge state into one of seven `WatchAction` outcomes
- [ ] `mergewatch.Classify` distinguishes `Merged`, `StillPending`, `Indeterminate`, `Conflict`, `RedCI`, `BranchProtectionChanged`, `HumanDisabledAutoMerge`, `TransientError{retriesLeft}` — verified by table-driven unit tests
- [ ] `mergewatch.Classify` exhausted-retry `TransientError` routes to `BranchProtectionChanged` semantics
- [ ] Coordinator's per-tick orchestration enumerates watched PRs by marker-first discovery or exactly-one eligible linked Looper PR with Looper-owned auto-merge enabled, then dispatches the returned WatchAction
- [ ] `Conflict` and `RedCI` actions apply only the Fixer Trigger label (configured by `roles.fixer.triggers.labels`) to the watched PR as the sole allowed Coordinator PR mutation; Coordinator performs no other PR mutation
- [ ] `BranchProtectionChanged` action removes `triaged`, `dispatch/*`, and stale role trigger labels from the linked Issue (re-Triage path) and removes the marker comment
- [ ] `HumanDisabledAutoMerge` action removes the marker comment silently — no other change, no comment posted, no re-Triage triggered
- [ ] `Indeterminate` (`mergeable=null` / `mergeable_state="unknown"`) has an independent `maxIndeterminateDuration` bound anchored by marker `first_unknown_at`; before the deadline it performs no label mutation and consumes no transient retry budget, after the deadline it routes using `BranchProtectionChanged` semantics
- [ ] `TransientError` includes 5xx, timeout, 429, and 403 secondary-rate-limit / abuse-detection responses, and carries `suggestedDelay` from `Retry-After` / `x-ratelimit-reset` where present
- [ ] `TransientError` with retries left edits the marker comment in place to update the retry counter and persist `next_retry_at`; future ticks before `next_retry_at` skip polling and do not decrement retries
- [ ] Marker comment lives on the linked Issue (not the PR) and uses `<!-- looper:coordinator:merge-watch pr=N head_sha=<sha> retries=N first_unknown_at=<RFC3339-or-empty> next_retry_at=<RFC3339-or-empty> -->` self-dedup format
- [ ] Marker comment is created on first observation requiring state, updated in place across ticks, unchanged on `StillPending`, updated on first `Indeterminate` to persist `first_unknown_at`, updated on transient errors to persist `next_retry_at`, removed on `Merged` / `HumanDisabledAutoMerge` / `BranchProtectionChanged`
- [ ] Marker comment updates are safe against overlapping ticks: per-Issue serialization prevents concurrent marker mutation for the same linked Issue; GitHub comment PATCH is not treated as CAS
- [ ] No new database tables, no schema migrations, no new persisted fields — Coordinator remains stateless per ADR-0001 (retry counter lives in the public marker comment body)
- [ ] Coordinator/Sweeper boundary preserved per ADR-0003: Coordinator's target of action is the Issue (label removal) or the Fixer Trigger label; Coordinator never closes Issues and never mutates PRs except the explicit Fixer Trigger label routing exception
- [ ] Unit-test matrix for `mergewatch` is fully covered including all eight WatchAction outcomes, including Indeterminate-before-deadline, Indeterminate-after-deadline, head-SHA reset, rate-limit `next_retry_at`, human-disabled cleanup discovered via marker, multi-PR ambiguity, and BranchProtectionChanged label-cleanup completeness
- [ ] Two `fake-gh` integration tests cover the Conflict path and the HumanDisabledAutoMerge path
- [ ] PR description **explicitly requests `@oracle` review** per AGENTS.md, citing ADR-0001, ADR-0003, ADR-0005
- [ ] PR description links PRD #352, slice #357, slice #358
- [ ] `go test ./...` passes; `go vet ./...` clean; `gofmt -l .` empty

## Blocked by

#358

### PR DESCRIPTION
**Pull request by mrcfps**
## Summary
- add a pure `internal/coordinator/mergewatch` classifier plus durable issue-marker handling for watched auto-merge PRs
- teach Coordinator to watch linked Looper-owned auto-merge PRs, route conflicts to Fixer, and send drift/exhausted watches back to re-triage
- extend GitHub/fake-gh support and add coordinator/config/API contract coverage for merge-watch state

## Why
Slice 3/4 for PRD #352 needs Coordinator to watch what happens after Reviewer opts a PR into GitHub-native auto-merge, while preserving ADR-0001 statelessness and ADR-0003 role boundaries.

## Authority / review request
This PR is authority-bearing. Please request `@oracle` review per AGENTS.md. The implementation preserves ADR-0001, ADR-0003, and ADR-0005 by rooting durable watch state on the linked Issue marker comment and limiting PR mutation to the configured Fixer trigger label.

## Links
- PRD #352
- Slice #357
- Slice #358
- Closes #359

## Validation
- `gofmt -l .`
- `LOOPER_CONFIG= go vet ./...`
- `LOOPER_CONFIG= go test ./...`

<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=worker · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

### CODE REVIEW

File: internal/coordinator/mergewatch_runner.go [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+	}
+	snapshot, tempErr, err := r.mergeWatchSnapshot(ctx, repo, cwd, issue.detail.Number, watchedPR, currentLogin)
+	if err != nil {
+		return false, err
+	}
+	if tempErr != nil {
+		snapshot.TemporaryError = tempErr
+	}
+	if !snapshot.HasLooperLabel || (snapshot.AutoMergeEnabled && !snapshot.AutoMergeOwnedByLooper) {
+		return false, r.deleteMergeWatchComment(ctx, repo, cwd, marker)
```
[the remark this answers is not in the document]
Comment (mrcfps) [reply internal/coordinator/mergewatch_runner.go]: @chatgpt-codex-connector This has been fixed in f995ecc.

Updated internal/coordinator/mergewatch_runner.go to classify transient merge-watch errors before the label-removal guard and preserve the existing marker head SHA when the transient snapshot is partial, so retry budget is consumed instead of deleting the watch marker. Added coordinator tests for both check-run and PR-detail transient failures.




<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=fixer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

File: internal/coordinator/mergewatch_runner.go [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+		case status != "completed" && requiredCheck(requiredChecks, nameKey):
+			checks.Pending = append(checks.Pending, checkRun.Name)
+		case mergeableState == "unstable" && requiredCheck(requiredChecks, nameKey) && (conclusion == "failure" || conclusion == "timed_out" || conclusion == "cancelled" || conclusion == "action_required" || conclusion == "startup_failure" || conclusion == "stale"):
+			checks.Failed = append(checks.Failed, checkRun.Name)
+		}
+	}
+	for requiredName := range requiredChecks {
+		if _, ok := seenChecks[requiredName]; !ok {
+			checks.Missing = append(checks.Missing, requiredName)
+		}
```
[the remark this answers is not in the document]
Comment (mrcfps) [reply internal/coordinator/mergewatch_runner.go:221]: @chatgpt-codex-connector This has been fixed in f995ecc.

Updated internal/infra/github/gateway.go and internal/coordinator/mergewatch_runner.go to load commit-status contexts alongside check runs and count required statuses as seen, pending, or failed during required-check evaluation. Added gateway and coordinator regression tests covering legacy required status contexts.




<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=fixer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

**Comment by mrcfps** [comment]
**Looper fixer round complete** — f995ecc

- ✅ Review comment on `internal/coordinator/mergewatch_runner.go` (@chatgpt-codex-connector) — [thread](https://github.com/nexu-io/looper/pull/422#discussion_r3272055524)
  - Updated internal/coordinator/mergewatch_runner.go to classify transient merge-watch errors before the label-removal guard and preserve the existing marker head SHA when the transient snapshot is partial, so retry budget is consumed instead of deleting the watch marker. Added coordinator tests for both check-run and PR-detail transient failures.
- ✅ Review comment on `internal/coordinator/mergewatch_runner.go:218` (@chatgpt-codex-connector) — [thread](https://github.com/nexu-io/looper/pull/422#discussion_r3272055531)
  - Updated internal/infra/github/gateway.go and internal/coordinator/mergewatch_runner.go to load commit-status contexts alongside check runs and count required statuses as seen, pending, or failed during required-check evaluation. Added gateway and coordinator regression tests covering legacy required status contexts.


<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=fixer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

File: internal/coordinator/runner.go [hunk 3]
```diff
@@ -177,6 +186,10 @@ func (r *Runner) DiscoverIssues(ctx context.Context, input DiscoveryInput) (Disc
 		}
 		loaded = append(loaded, issue)
 	}
+	mergeWatchRetriggers, err := r.applyMergeWatch(ctx, input.Repo, project.RepoPath, loaded, config.ProjectRoleConfigs(*r.config, input.ProjectID))
+	if err != nil {
+		return DiscoveryResult{}, err
+	}
 
 	deps, err := r.buildDependencyState(ctx, input.Repo, project.RepoPath, loaded, triageCfg, dispatchCfg, roleCfg.Dependencies)
 	if err != nil {
```
[the remark this answers is not in the document]
Comment (mrcfps) [reply internal/coordinator/runner.go]: @chatgpt-codex-connector This has been fixed in 956d87c.

Filtered merge-watch retriggers out of same-tick dependency and dispatch processing in internal/coordinator/runner.go, and added TestRunnerMergeWatchRetriggerSkipsSameTickDispatch to lock the cleanup behavior.




<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=fixer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

File: internal/coordinator/mergewatch_runner.go [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+		if isTransientMergeWatchError(err) {
+			return mergewatch.PRSnapshot{Repo: repo, PRNumber: prNumber, IssueNumber: issueNumber, HeadSHA: detail.HeadSHA, AutoMergeEnabled: detail.AutoMerge != nil, AutoMergeOwnedByLooper: detail.AutoMerge != nil && strings.EqualFold(detail.AutoMerge.EnabledBy, currentLogin), Mergeable: detail.Mergeable, MergeableState: detail.MergeableState}, &mergewatch.TemporaryError{SuggestedDelay: time.Minute}, nil
+		}
+		return mergewatch.PRSnapshot{}, nil, err
+	}
+	checks := mergewatch.RequiredCheckSummary{}
+	mergeableState := strings.ToLower(strings.TrimSpace(detail.MergeableState))
+	protection, err := r.github.GetBranchProtection(ctx, githubinfra.BranchProtectionInput{Repo: repo, Branch: detail.BaseRefName, CWD: cwd})
+	if err != nil {
+		return mergewatch.PRSnapshot{}, nil, err
```
[the remark this answers is not in the document]
Comment (mrcfps) [reply internal/coordinator/mergewatch_runner.go:188]: @chatgpt-codex-connector This has been fixed in 956d87c.

Handled transient GetBranchProtection failures as merge-watch retryable errors in internal/coordinator/mergewatch_runner.go, with regression coverage in TestRunnerMergeWatchBranchProtectionTransientErrorConsumesRetryBudget.




<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=fixer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

**Comment by mrcfps** [comment]
**Looper fixer round complete** — 956d87c

- ✅ Review comment on `internal/coordinator/runner.go` (@chatgpt-codex-connector) — [thread](https://github.com/nexu-io/looper/pull/422#discussion_r3272184306)
  - Filtered merge-watch retriggers out of same-tick dependency and dispatch processing in internal/coordinator/runner.go, and added TestRunnerMergeWatchRetriggerSkipsSameTickDispatch to lock the cleanup behavior.
- ✅ Review comment on `internal/coordinator/mergewatch_runner.go:188` (@chatgpt-codex-connector) — [thread](https://github.com/nexu-io/looper/pull/422#discussion_r3272184317)
  - Handled transient GetBranchProtection failures as merge-watch retryable errors in internal/coordinator/mergewatch_runner.go, with regression coverage in TestRunnerMergeWatchBranchProtectionTransientErrorConsumesRetryBudget.


<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=fixer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>
